File: src/python/input.py
import os.path
from typing import Dict

def get_input(path: str) -> str:
   with open(path, 'r', encoding="utf8") as f:
      lines = f.read()
   return lines.strip()

def get_answers(path: str) -> Dict[str, str]:
   with open(path, 'r', encoding="utf8") as f:
      lines = [x.replace('\n', '') for x in f.readlines()]
   return { 'test1': lines[0], 'part1':  lines[1], 'test2': lines[2], 'part2': lines[3]}

File: src/go/input.go
package core

import (
	"bufio"
	"os"
	"strings"
)

type Answers struct {
	test1 string
	test2 string
	part1 string
	part2 string
}

// read text file and return the content
func GetInput(path string) string {
	f, _ := os.ReadFile(path)
	return strings.TrimRight(string(f), "\n")
}

func GetAnswers(path string) Answers {
	f, _ := os.Open(path)
	scanner := bufio.NewScanner(f)

	answers := Answers{}

	scanner.Scan()
	answers.test1 = scanner.Text()
	scanner.Scan()
	answers.part1 = scanner.Text()
	scanner.Scan()
	answers.test2 = scanner.Text()
	scanner.Scan()
	answers.part2 = scanner.Text()

	return answers
}
